Subscription/Subscription.sol

```diff
@@ -173,11 +181,6 @@ contract Subscription {
         uint256 balance = ERC20(tokenAddress).balanceOf(from);
 
         return (
-            ( requiredToAddress == address(0) || to == requiredToAddress ) &&
-            ( requiredTokenAddress == address(0) || tokenAddress == requiredTokenAddress ) &&
-            ( requiredTokenAmount == 0 || tokenAmount == requiredTokenAmount ) &&
-            ( requiredPeriodSeconds == 0 || periodSeconds == requiredPeriodSeconds ) &&
-            ( requiredGasPrice == 0 || gasPrice == requiredGasPrice ) &&
```

Comment:
removed these checks because they are performed within the `getSubscriptionHash` call on ln176.